src/texture.cr

```diff
@@ -62,17 +66,26 @@ module SDL
       blend_mode
     end
 
-    #def update
-    #end
-
-    #def update_yuv
-    #end
-
-    #def lock
-    #end
+    # def update
+    # end
+
+    # def update_yuv
+    # end
+
+    def lock
+      begin
+        pitch = uninitialized Int32
+        pixels_pointer = uninitialized Void*
+        ret = LibSDL.lock_texture(@texture, nil, pointerof(pixels_pointer), pointerof(pitch))
+        raise Error.new("SDL_LockTexture") unless ret == 0
+        yield Slice.new(Pointer(UInt32).new(pixels_pointer.address), width * height), pitch
+      ensure
+        LibSDL.unlock_texture(@texture)
+      end
+    end
```

Comment:
In Crystal we usually have a `lock()` and `unlock()` pair of methods, then an override with a block:

```crystal
def lock : {Pointer(Slice(UInt32)), Int32}
  pitch = uninitialized Int32
  pixels_pointer = uninitialized Void*
  ret = LibSDL.lock_texture(self, nil pointerof(pixels_pointer), pointerof(pitch))
  raise Error.new("SDL_LockTexture") unless ret == 0
  {Slice.new(Pointer(UInt32).new(pixels_pointer.address), width * height), pitch}
end

def unlock : Nil
  LibSDL.unlock_texture(self)
end

def lock(& : Pointer(Slice(UInt32)), Int32 ->) : Nil
  buffer, pitch = lock
  begin
    yield buffer, pitch
  ensure
    unlock
  end
end
```